core/node/da_clients/src/eigen/verifier.rs

```diff
@@ -0,0 +1,699 @@
+            Ethereum,
+        > = RootProvider::new_http(url);
+
+        let svc_manager_addr = alloy::primitives::Address::from_str(&cfg.svc_manager_addr)
+            .map_err(|_| VerificationError::ServiceManagerError)?;
+
+        let eigenda_svc_manager = EigenDAServiceManager::new(svc_manager_addr, provider);
+        Ok(Self {
+            kzg,
+            eigenda_svc_manager,
+            cfg,
+        })
+    }
+
+    /// Return the commitment from a blob
+    fn commit(&self, blob: Vec<u8>) -> Result<G1Affine, VerificationError> {
+        let blob = Blob::from_bytes_and_pad(&blob.to_vec());
+        self.kzg
+            .blob_to_kzg_commitment(&blob, PolynomialFormat::InEvaluationForm)
+            .map_err(|_| VerificationError::KzgError)
+    }
+
+    /// Compare the given commitment with the commitment generated with the blob
+    pub fn verify_commitment(
+        &self,
+        expected_commitment: G1Commitment,
+        blob: Vec<u8>,
+    ) -> Result<(), VerificationError> {
+        let actual_commitment = self.commit(blob)?;
+        let expected_commitment = G1Affine::new_unchecked(
+            Fq::from(num_bigint::BigUint::from_bytes_be(&expected_commitment.x)),
+            Fq::from(num_bigint::BigUint::from_bytes_be(&expected_commitment.y)),
+        );
+        if actual_commitment != expected_commitment {
+            return Err(VerificationError::WrongProof);
+        }
+        Ok(())
+    }
+
+    fn hash_encode_blob_header(&self, blob_header: BlobHeader) -> Vec<u8> {
+        let mut blob_quorums = vec![];
+        for quorum in blob_header.blob_quorum_params {
+            let quorum = Token::Tuple(vec![
+                Token::Uint(ethabi::Uint::from(quorum.quorum_number)),
+                Token::Uint(ethabi::Uint::from(quorum.adversary_threshold_percentage)),
+                Token::Uint(ethabi::Uint::from(quorum.confirmation_threshold_percentage)),
+                Token::Uint(ethabi::Uint::from(quorum.chunk_length)),
+            ]);
+            blob_quorums.push(quorum);
+        }
+        let blob_header = Token::Tuple(vec![
+            Token::Tuple(vec![
+                Token::Uint(ethabi::Uint::from_big_endian(&blob_header.commitment.x)),
+                Token::Uint(ethabi::Uint::from_big_endian(&blob_header.commitment.y)),
+            ]),
+            Token::Uint(ethabi::Uint::from(blob_header.data_length)),
+            Token::Array(blob_quorums),
+        ]);
+
+        let encoded = encode(&[blob_header]);
+
+        let mut keccak = Keccak::v256();
+        keccak.update(&encoded);
+        let mut hash = [0u8; 32];
+        keccak.finalize(&mut hash);
+        hash.to_vec()
+    }
+
+    fn process_inclusion_proof(
+        &self,
+        proof: &[u8],
+        leaf: &[u8],
+        index: u32,
+    ) -> Result<Vec<u8>, VerificationError> {
+        let mut index = index;
+        if proof.len() == 0 || proof.len() % 32 != 0 {
+            return Err(VerificationError::WrongProof);
```

Comment:
Would be good to differentiate the errors since the `VerificationError::WrongProof` is used in many places.